packages/rum/src/domain/record/serialize.ts

```diff
@@ -232,8 +243,9 @@ function serializeCDataNode(): CDataNode {
 
 export function serializeChildNodes(node: Node, options: SerializeOptions): SerializedNodeWithId[] {
   const result: SerializedNodeWithId[] = []
+  const childNodes = isExperimentalFeatureEnabled('recordShadowDom') ? getChildNodes(node) : node.childNodes
```

Comment:
I had the same thought and went against it but mostly because I think helper should be pure and should not have side effect (aka FF). Happy to change it if you prefer